clippy_lints/src/assertions_on_result_states.rs

```diff
@@ -96,3 +100,15 @@ fn has_debug_impl<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) -> bool {
         .get_diagnostic_item(sym::Debug)
         .map_or(false, |debug| implements_trait(cx, ty, debug, &[]))
 }
+
+fn is_unit_type(ty: Ty<'_>) -> bool {
```

Comment:
Can't you call `ty.is_unit() || ty.is_never()` instead of this function?

Reply:
Good idea. Was copying it from another older lint before.

Reply:
Oh which one? We should also fix it there... (not in this PR though.

Reply:
Here: https://github.com/rust-lang/rust-clippy/blob/7c427f0fdadf07c117d23d0b3352a4870c9b17d3/clippy_lints/src/map_unit_fn.rs#L99

Reply:
Are you motivated to fix it also there and open a PR for it? :)

Reply:
Sure, will fix it tomorrow.